solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/SolrZkClient.java

```diff
@@ -993,14 +1051,33 @@ public void updateACLs(final String root) throws KeeperException, InterruptedExc
         ZkMaintenanceUtils.VISIT_ORDER.VISIT_POST,
         path -> {
           try {
-            setACL(path, getZkACLProvider().getACLsToAdd(path), true);
-            log.debug("Updated ACL on {}", path);
+            runWithCorrectThrows(
+                "updating acls", () -> client.setACL().withACL(null).forPath(path));
           } catch (NoNodeException ignored) {
             // If a node was deleted, don't bother trying to set ACLs on it.
           }
         });
   }
 
+  @FunctionalInterface
+  protected interface SupplierWithException<T> {
+    T get() throws Exception;
+  }
+
+  protected <T> T runWithCorrectThrows(String action, SupplierWithException<T> func)
+      throws KeeperException, InterruptedException {
```

Comment:
but we catch InterruptedException now

Reply:
We catch it, and re-throw it. The point is to not cover it up in a RuntimeException.